### ISSUE TITLE
Next feature release

### ISSUE THREAD

**Issue by Lineflyer** [cgeo/cgeo#6956]
Here we are once again. Lets chat about the next possible feature release and the ToDo before we can release.

The main change in our `master` branch is the promotion of the new mapsforge implementation. This also means, that the known bugs should get some attention and we should try to classify them (i.e. which bug is not OK for going live).
I took a look at the list of bugs labeled with "New Map": https://github.com/cgeo/cgeo/issues?q=is%3Aopen+sort%3Aupdated-desc+label%3A%22New+Map%22+label%3ABug

There are 10 issues currently on that list, where my personal impression is, that the following should be solved or at least taken a closer look before release:
- [ ] #5902 <-- Regression of info shown on the map, for me this is a must solve issue
- [ ] #5900 <-- Regression regarding waypoints on map
- [x] #6395 <-- Regression regarding waypoints on map
- [x] #6984 <-- Map zooming out on rotation

The other listed bugs are IMHO not nice, but also not critical and could be solved at a later point of time. Still if something else on the list is also being solved, it wont harm.

Besides that, we also have one regression in terms of features not yet implemented on the new map but exisiting in the old map: #5885 (1/4 mile circles around caches). I would however consider this as a minor regression which can be fixed later.

@rsudev 
Did I miss something important when it comes to bugs or unimplemented features of the old map implementation?

**Comment by Matze2** [comment]
For me #6933 was the reason to go back to old v3 version. It makes the whole map display completely unreliable.

**Comment by Lineflyer** [comment]
@pstorch @rsudev Any comments or contribution from your side would be appreciated :)

**Comment by pstorch** [comment]
Sorry for the late reply. I tried to get into the marker and overlay logic of the new map,  got lost. Maybe it's easier for @rsudev.

**Comment by Lineflyer** [comment]
@Matze2 
> For me #6933 was the reason to go back to old v3 version. It makes the whole map display completely unreliable.

I did not list this issue as a "must solve" as it involves a map switch.
Why is this a "killer" for you? Are you frequently switching maps (which I consider a rare case) or something else?

**Comment by Lineflyer** [comment]
Question:
Do we want to get more feedback regarding the mapsforge version switch and the new logging page by having a beta version out?
I would appreciate it, however this means we would merge `master` to `release`for that. In this case we should be able to revert (or use a backup branch) in case of urgent releases. I guess this is no problem on Github...is it?

**Comment by kumy** [comment]
> we should be able to revert (or use a backup branch) in case of urgent releases

This is the purpose of tags :)

**Comment by Lineflyer** [comment]
> This is the purpose of tags :)

So lets wait for #6977 to be merged (and maybe some other fixes for the items above as they come) and then produce a beta version.

**Comment by Matze2** [comment]
I tested a little bit with the new map this weekend.

The most annoying thing is not the "map switch" issue but the "zoom out" issue.

At least on device rotation (but I think also on other redraw events) the map zooms out, e.g from 1km to 10km. The workaround is to disable device rotation.

**Comment by Lineflyer** [comment]
> At least on device rotation (but I think also on other redraw events) the map zooms out, e.g from 1km to 10km. The workaround is to disable device rotation when using new maps.

Indeed, can reproduce.
Is this covered in any issue here yet?

**Comment by Matze2** [comment]
I don't think so. I mentioned it in #6933. I can create it.

**Comment by Lineflyer** [comment]
Yes, please.

**Comment by Lineflyer** [comment]
Added #6984 above

**Comment by Lineflyer** [comment]
I think we meanwhile ensured, that the new logging page implementation as well as the new mapsforge implementation is up and running on `master` and the remaining issues are documented at least.

Any objections to merge `master`  into `release` with the first interim target to produce a beta version (RC) for our beta testers?

**Comment by mucek4** [comment]
Question is, shall we go now, just before the webpage change?

**Comment by Lineflyer** [comment]
I am undecided about this.
On one hand potential adaptions would then base on the newest code...on the other hand, the current release is quite stable, whereas ˋmasterˋ is for sure more error prone.
But: The new log page is an important thing.

**Comment by Lineflyer** [comment]
I have to come back to this once again after sleeping a night over it:
I think there are more advantages to have ˋmasterˋ out in the field as soon as possible and also base all upcoming needed changes on this. Otherwise we would have to work on very different branches in parallel.

Seeing that @rsudev started working on some of the "New map" topics, gives me even more confidence, that ˋmasterˋ will be suitable for releasing it rather soon.

**Comment by pstorch** [comment]
+1 for pushing this out into the field (Beta first).

**Comment by Lineflyer** [comment]
Given that @rsudev is working on quite some of the map issues (highly appreciated btw) the question is, if we wait for those upcoming changes or merge to `release` now.
Merging now has the benefit that we get more clues from a beta version, the question to @rsudev would be if it is OK for him to continue working against `release` right now (probably it does not matter thanks to Github?!).

**Comment by rsudev** [comment]
Before that, I probably should sync the translations once more. In principle we have branches on corwdin as well, but that doesn't work as smooth as hoped for and untangle the release branch over there I'd rather do unhurriedly.
What about a beta for the weekend (like Friday evening)?

**Comment by Lineflyer** [comment]
Friday sounds good in principle if someone else than me can publish it, as I am on tour then without PC access.
Else I could offer to do that until today afternoon if you manage to import Crowdin and merge branches today.

**Comment by rsudev** [comment]
I am about to bring in the translations. I have added Greek already (which IMO is uncontroversial), but this leaves open Arabic, Filipino, Tagalog and Indonesian. They are translated sufficiently according to our criteria, but likely not backed by a geocacher/geocaching community (similar to Cebuano, which we already included in master).
All of these translations have likely been driven by the platform 'utopian' as you might remember, not necessarily by genuine interest into geocaching or our app. That of course doesn't preclude quality and passion in the translation as such (I had e.g. that impression esp. from the translator to Arabic).
For the Beta we can surely first leave it as it is now, but we should reconsider and decide before release.

**Comment by Matze2** [comment]
I don't want to interrupt your crowdin discussions. I just want to give short input about the maturity of the new maps feature.

At the beginning I was very skeptical about the new Map but now I start to like it.
Not only it allows to use appealing offline maps like Freizeitkarte it contains also another "feature" for base members which might be just there by accident.

It now shows "more often" PMO caches on the live map. Base members which also have access to PMO coordinates can now find candidates much easier where it is worth to "organize" the coordinates.

I was always interested in this feature since the rough position is known in the nearby list - so why not make it available.

Thank you all for the big effort to make the new Maps really working! 👍

**Comment by SammysHP** [comment]
Cache parsing is not affected by the new map.

**Comment by Lineflyer** [comment]
> It would be nice, if we could merge #7025 before.

Thats merged now.
Any objections now...or who can do the merge?

**Comment by Lineflyer** [comment]
I am OK for merging and issueing a beta version asap.
If someone finds time to merge....feel free.

**Comment by UniQP** [comment]
I did the merge.

**Comment by Lineflyer** [comment]
I built an RC version and will now smoke test it on my device first.

**Comment by Lineflyer** [comment]
@rsudev 
I opened the following issues (maybe related to the new map) based on Google Play feedback for the beta version. If you find time to take a look, that would be great:
- [ ] #7032 
- [ ] #7033
- [ ] #7034

**Comment by Lineflyer** [comment]
FYI:
I updated the existing beta version by a new version 2018.06.11-RC, which includes latest fixes as mentioned above.

**Comment by Lineflyer** [comment]
@rsudev No new information, significant crashes or user reports from the beta version.
What do you think: Any chance you will find time to look at the two open issue (listed in the initial posting here) or shall we release and fix them later?
I don't think there is a hurry to release, but still the new version would also bring the new logging page which is desirable.

**Comment by Lineflyer** [comment]
FYI:
Due to the changes from #7063 a new RC version has been just published to the beta channel.

**Comment by Lineflyer** [comment]
As I think we need to make a step forward I would suggest to release the current beta version.
There are some regressions when it comes to the mapsforge maps, but as I did not see a reaction from @rsudev, who might have no time to work on this, its not worth to just wait.

Any objection?

**Comment by Lineflyer** [comment]
No reaction the last two days. If no one stops me I will release a new version this night.
The remaining "ToDo" can be worked around by not using the new mapsforge library...so I do not consider them critical, but will maybe add another FAQ entry for the time being.

**Comment by Lineflyer** [comment]
New release 2018.06.26 has been smoke tested and uploaded.
Additionally FAQ entries have been added for the mapsforge regressions ans linked from changelog.

To Do:
- [x] Check/Insert correct version name as headline in ```changelog-release.xml```
- [x] Build release on CI server
- [x] Perform smoke test with resulting ```cgeo-release.apk``` before continuing
- [x] Upload and publish ```cgeo-release.apk``` to Google Play
- [ ] Tag the release on Github
- [ ] Publish the release on Github including upload of ```cgeo-release.apk``` to Github
- [ ] Publish release on F-Droid 
- [ ] Rename/Create milestones on Github
- [ ] Trigger notification to status.cgeo.org via CI server once the release is available on Google Play
- [ ] Post release info to Facebook 
- [ ] Post release info to G+ 
- [ ] Post release info to Twitter (via @SammysHP)

Help with tagging the release and further steps is appreciated.

**Comment by Lineflyer** [comment]
Publish to F-Droid fails, see: http://ci.cgeo.org/view/Fdroid/job/Publish%20FDroid%20-%20RELEASE/35/console

**Comment by Lineflyer** [comment]
Thanks @kumy for taking care of F-Droid. We are done with release.
Thank you all and please be reminded to the open regressions (which already got reported by our users as feedback to this update).

I will open a new issue to track those regression issues for "next bugfix release".
